Blog post for pulumi-go-provider-v1 (#15091)
pulumi/docs: content/blog/pulumi-go-provider-v1/index.md

---
title: "Pulumi Go Provider SDK v1.0: Multi-Language Infrastructure Integration for Modern Teams"
date: 2025-05-16
updated: 2025-05-20
draft: false
allow_long_title: true

meta_desc: "Pulumi Go Provider SDK v1.0: Build custom infrastructure providers in hours, not weeks, while unlocking cross-team collaboration and standardization"
meta_image: meta.png
authors:
    - eron-wright
    - meagan-cojocar
tags:
    - releases 
    - platform-teams
    - features
    - iac
    - components
---

> At Pulumi, we are committed to accelerating your cloud infrastructure journey by eliminating barriers between your teams and the tools they need. Today, we're thrilled to announce a game-changing milestone that puts unprecedented power in your hands: the **v1.0.0 release of the Pulumi Provider SDK** for Go!

Reviewer: see naming comment below ... this is actually a "third variation" of the name !

<!--more-->

## Unlock the Full Potential of Your Cloud Ecosystem

> The Pulumi Go Provider SDK v1.0 is a transformative framework that enables you to integrate **any** service, tool, or API into your infrastructure as code workflows. After extensive development since 2022 and battle-testing with enterprise customers, this framework is now **stable and production-ready** for your most critical infrastructure needs.

Reviewer: ```suggestion
The Pulumi Go Provider SDK v1.0 is a transformative framework that enables you to integrate **any** service, tool, or API into your infrastructure as code workflows. After extensive development and testing with customers, this framework is now **stable and production-ready** for your most critical infrastructure needs.
```

Reviewer: in general, if we can avoid "war metaphors" (battle cards, war rooms, etc), so much the better

Reviewer: if you have a compelling example of a service, tool, or API that customers wanted to integrate, it would be great to note that up front ... otherwise, the post is a little more "abstract."

		integration.WithProvider(provider()),
	)
	require.NoError(t, err)

	// Test the lifecycle methods of the HelloWorld resource.
	integration.LifeCycleTest{
		Resource: "pkg:index:HelloWorld",
		Create: integration.Operation{
			Inputs: property.NewMap(map[string]property.Value{
				"name": property.New("tester"),
				"loud", property.New(true),
			}),
			Hook: func(inputs, output property.Map) {
				message := output.Get("message").AsString()
				assert.Equal(t, "Hello, tester", message)
			},
		},
	}.Run(t, server)
}
```

Need help debugging your provider during development? Our [Debugging Providers guide](/docs/iac/using-pulumi/extending-pulumi/debugging-providers/) walks you through common troubleshooting techniques and tools.

## Standardize Infrastructure with Powerful Component Resources

Beyond simple resources, providers can also define **Component Resources**. Components are higher-level abstractions that encapsulate multiple child resources working together to implement a specific capability. They are perfect for codifying organizational standards and best practices into reusable (cross-language!) building blocks.

> The Pulumi Provider SDK v1.0.0 introduces support for defining components using a **functional style**, similar to how you might define a local component within a single program. This provides a streamlined way to build components. Here's an example using the functional style:

Reviewer: ```suggestion
The Pulumi Go Provider SDK introduces support for defining components using a **functional style**, similar to how you might define a local component within a single program. This provides a streamlined way to build components. Here's an example using the functional style:
```

Reviewer: In general, I'm confused on whether you want to call this the "Go Provider SDK" or drop the word "Go" ... this draft is about 50/50!   My suggestion is pick one, use it everywhere, and drop the version (since 1.1 will have all the same attributes, etc.)

## Get Started Today!

> The GA of the Pulumi Go Provider SDK represents a significant step forward in simplifying and accelerating the development of Pulumi providers. By focusing on a code-first approach, automatic schema generation, and providing a robust testing framework, we are empowering engineers to integrate Pulumi with virtually any service or tool.

Reviewer: ```suggestion
The release of the Pulumi Go Provider SDK represents a significant step forward in simplifying and accelerating the development of Pulumi providers. By focusing on a code-first approach, automatic schema generation, and providing a robust testing framework, we are empowering engineers to integrate Pulumi with virtually any service or tool.
```
